impeller/geometry/vector.h

```diff
@@ -288,6 +288,43 @@ struct Vector4 {
 static_assert(sizeof(Vector3) == 3 * sizeof(Scalar));
 static_assert(sizeof(Vector4) == 4 * sizeof(Scalar));
 
+struct HalfVector4 {
```

Comment:
We could convert Vector3 and others to be TVector3 with template types for floats or half-floats. Then `using Vector3 = TVector<Scalar>;` and `using HalfVector3 = TVector<Half>`. Will significantly dry up this code.